File: domain/src/main/java/hanghae/domain/entity/MovieTheaterEntity.java
package hanghae.domain.entity;

import jakarta.persistence.*;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "movie_theater")
@NoArgsConstructor
public class MovieTheaterEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long movieTheaterId;

    @ManyToOne
    @JoinColumn(name = "movie_id")
    private MovieEntity movie;

    @ManyToOne
    @JoinColumn(name = "theater_id")
    private TheaterEntity theater;
}

File: module-adapter/src/main/kotlin/com/example/redis/movie/jpa/MovieTheaterEntity.kt
package com.example.redis.movie

import com.example.redis.BaseEntity
import com.example.redis.theater.SeatEntity
import com.example.redis.theater.TheaterEntity
import jakarta.persistence.*
import org.hibernate.annotations.BatchSize

@Table(name = "movie_theater")
@Entity
class MovieTheaterEntity(

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id")
    val movie: MovieEntity,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id")
    val theater: TheaterEntity,

    @BatchSize(size = 1_000)
    @OneToMany(mappedBy = "movieTheater", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    val seats: MutableList<SeatEntity> = mutableListOf(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    ): BaseEntity() {
}